### PR TITLE
added control for exponential decay learning rate schedule

### PR DESCRIPTION
**Pull request by john-waczak**

### CODE REVIEW

**Comment by ablaom** [comment]
Many thanks! I appreciate it takes some work to wrap your head around a new interface. 

Would you be willing to add a test? You can probably use the [cyclic learning rate test](https://github.com/john-waczak/MLJIteration.jl/blob/33d2cbdde869b05ec8fd68e9f72d9516ec9917ba/test/controls.jl#L169) as a template. The data `X, y` is defined at the top of test/controls.jl and you can probably use `DummyModel` (defined and explained [here](https://github.com/JuliaAI/MLJIteration.jl/blob/33d2cbdde869b05ec8fd68e9f72d9516ec9917ba/test/_dummy_model.jl#L32)) without understanding too much about what it does. It's just a model with a `learning_rate` hyper-parameter.
